core/pkg/server/sender.go

```diff
@@ -214,7 +215,13 @@ func (s *Sender) Do(allWork <-chan runwork.Work) {
 	defer s.logger.Reraise()
 	s.logger.Info("sender: started", "stream_id", s.settings.RunId)
 
+	hangDetectionInChan := make(chan runwork.Work, 32)
```

Comment:
a little bit more documentation would be helpful here

Reply:
what do you think of the documentation on warnOnLongOperations?